c/parallel/test/test_for.cpp

```diff
@@ -115,8 +115,9 @@ C2H_TEST("for works with stateful operators", "[for]")
     "op",
     R"XXX(
 struct invocation_counter_state_t { int* d_counter; };
-extern "C" __device__ void op(invocation_counter_state_t* state, int* a) {
-  atomicAdd(state->d_counter, *a);
+extern "C" __device__ void op(void* state_ptr, void* a_ptr) {
+  invocation_counter_state_t* state = static_cast<invocation_counter_state_t*>(state_ptr);
+  atomicAdd(state->d_counter, *static_cast<int*>(a_ptr));
```

Comment:
Since operation definitions in other tests use typed pointers, perhaps changes here are not necessary?